### PR TITLE
Initial commit of structures to support a configuration service

### PR DESCRIPTION
**Pull request by autodidaddict**
Signed-off-by: Kevin Hoffman <autodidaddict@users.noreply.github.com>

### CODE REVIEW

File: configservice/config-service.smithy [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/// real-time configuration change notifications on the "push" topic. It's noteworthy
+/// that this configuration profile does NOT contain any connection information for NATS,
+/// as allowing that to change at runtime would be a security and reliability risk
+structure HostConfigurationProfile {
+    /// The list of capability providers a host should automatically start
+    autoStartProviders: ProviderReferenceList
+
+    /// A set of credentials the host can use for fetching artifacts
+    registryCredentials: RegistryCredentialMap
+}
```
Comment (brooksmtownsend) [inline configservice/config-service.smithy:37]: Not sure if this is just the starting point, but for a configuration profile I'd love to be able to start actors, providers, and establish link definitions

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]

**Review by brooksmtownsend — APPROVED** [review]

**Review by thomastaylor312 — APPROVED** [review]

File: configservice/config-service.smithy [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+structure ProviderReference {
+    imageReference: String,
+    linkName: String
+}
+
+/// A set of credentials to be used for fetching from specific registries
+map RegistryCredentialMap {
+    /// The key of this map is the OCI/BINDLE URL without the artifact reference. Credentials
```
[the remark this answers is not in the document]
Comment (autodidaddict) [reply configservice/config-service.smithy:57]: so far I've been assuming that if a bindle reference is prefixed with `bindle://` then that's the thing that I then use to look up credentials.

So if you pass a bindle reference of `bindle://foo.bar.baz` then that string is the key in the hash map which is used to look up credentials.

How is this impacted by bindles not being URL-addressable?

[the remark this answers is not in the document]
Comment (autodidaddict) [reply configservice/config-service.smithy:57]: @thomastaylor312 this is designed to allow us to look up credentials by the configured equivalent of the `BINDLE_URL` environment variable.. Maybe we'll need to :pear: on the implementation of the lookup of bindle creds

[the remark this answers is not in the document]
Comment (thomastaylor312) [reply configservice/config-service.smithy:57]: Isn't the idea here to support multiple possible servers (OCI or bindle)? If we assume only one bindle server, then you are correct `bindle://` could easily be used. If we want to support multiple, we need something a bit more

[the remark this answers is not in the document]
Comment (autodidaddict) [reply configservice/config-service.smithy:57]: the idea is absolutely to support multiple servers. Each server is a key in the map.

**Pull request merged**